testing/test_script.py

```diff
@@ -1,18 +1,250 @@
+              "_T2w_lesion-manual.nii.gz"
+    subprocess.check_output(command, shell=True)
+
+    # populate derivatives for sub-test003
+    command = "cp " + derivatives + "sub-unf01/anat/sub-unf01_T2w_seg-manual.nii.gz " + \
+              derivatives + "sub-test003/anat/sub-test003" + \
+              "_T2w_seg-manual.nii.gz"
+    subprocess.check_output(command, shell=True)
+
+    command = "cp " + derivatives + "sub-unf01/anat/sub-unf01_T2w_lesion-manual.nii.gz " + \
+              derivatives + "sub-test003/anat/sub-test003" + \
+              "_T2w_lesion-manual.nii.gz"
+    subprocess.check_output(command, shell=True)
+
+    # Model needs to be inside the log_directory since we use a config file.
+    command = "cp testing_data/model_unet_test.pt testing_script/best_model.pt"
+    subprocess.check_output(command, shell=True)
+
+    list1 = ["sub-test002"]
+    list2 = ["sub-test003"]
+
+    # add subjects to participants.tsv
+    append_list_as_row("testing_data/participants.tsv", list1)
+    append_list_as_row("testing_data/participants.tsv", list2)
+
+
+def test_testing_with_uncertainty():
+    # Test config. Uses Uncertainty
+    subprocess.check_output(["ivadomed -c testing_data/model_config_test.json"], shell=True)
+
+
+def test_training():
+    # Train config 
+    subprocess.check_output(["ivadomed -c testing_data/model_config.json"], shell=True)
+
+
+# def test_training_curve():
+# using the results from previous training
+#   command = "ivadomed_training_curve -i testing_script/ -o training"
+#  subprocess.check_output(command, shell=True)
+
+
+def test_create_eval_json():
+    # modify train config
+    command = "cp testing_data/model_config.json testing_data/model_config_eval.json"
+    subprocess.check_output(command, shell=True)
+    file_conf = open("testing_data/model_config_eval.json", "r")
+    initial_config = json.load(file_conf)
+    file_conf.close()
+    file_conf = open("testing_data/model_config_eval.json", "w")
+    initial_config["command"] = "eval"
+    initial_config["transformation"] = {
+        "Resample": {
+            "wspace": 0.75,
+            "hspace": 0.75
+        },
+        "RandomAffine": {
+            "degrees": 4.6,
+            "translate": [0.03, 0.03],
+            "scale": [0.98, 1]
+        },
+        "NumpyToTensor": {},
+        "NormalizeInstance": {
+            "applied_to": ["im"]
+        }}
+    initial_config["testing_parameters"] = {
+        "binarize_prediction": False,
+        "uncertainty": {
+            "epistemic": False,
+            "aleatoric": False,
+            "n_it": 2
+        }}
+    json.dump(initial_config, file_conf)
+
+
+def test_eval():
+    subprocess.check_output(["ivadomed -c testing_data/model_config_eval.json"], shell=True)
+
+
+def test_create_automate_training_json():
+    # modify train config
+    command = "cp testing_data/model_config_eval.json testing_data/model_config_auto.json"
```

Comment:
Yeah but also, this came after the first release of data-testing. Basically, the idea is to store a low number of file and modify them each time. otherwise each time you need a new test, or modify the config file, you will need to modify like 10 files and do a release in data-testing (+ change link in ivadomed_download_data)... So I am trying to figure out a way of doing our maintenance at one single place :)